src/defineMacro.js

```diff
@@ -116,3 +128,24 @@ export const _macros: MacroMap = {};
 export default function defineMacro(name: string, body: MacroDefinition) {
     _macros[name] = body;
 }
+
+export const _conditionals: {[string]: true} = {};
+
+export function defineConditional(
+    name: string,
+    evaluate: MacroContextInterface => boolean
+) {
+    _conditionals[name] = true;
+    // When an \if... is expanded, TeX reads ahead as far as necessary
+    // to determine whether the condition is true or false; and if false,
+    // it skips ahead until finding the \else, \or, or \fi that ends the
+    // skipped text.
+    defineMacro(name, function(context) {
+        const condition = evaluate(context);
+        context.conditions.push(condition);
+        if (!condition) {
```

Comment:
I believe this is the only place where `conditions`' boolean values are used.  Which suggests that you just need a depth counter, not an actual stack, because here it's known from line 144.